Sources/MCP/Server/Tools.swift

```
@@ -260,3 +260,36 @@ public enum CallTool: Method {
 public struct ToolListChangedNotification: Notification {
     public static let name: String = "notifications/tools/list_changed"
 }
+
+// MARK: - Type-Safe Parameter Parsing
+
+/// Protocol for types that can be parsed from MCP tool arguments
+/// This is typically implemented by structs decorated with @ToolRegistration, @Schema, etc.
+public protocol MCPParameterParsable {
```

Consider making the parsing API throwing (or returning a Result) to carry diagnostics instead of only returning nil. This helps tools show actionable error messages to users.